### PR TITLE
suggested improvements

### PR DESCRIPTION
**Pull request by craicoverflow**
- Fixed Markdown warnings (I recommend [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) VS Code extension.
- Added more paragraphs.
- Grammar suggestions.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -174,83 +192,101 @@ app.get('/users', (req, res) => {
     res.json(users);
 });
 ```
-- After adding new routes and restarting the server, visit localhost:4000/users or localhost:4000/tasks from your browser! Or simply,
-in your console type:
+
+After adding new routes and restarting the server, visit localhost:4000/users or localhost:4000/tasks from your browser! Or in your console, simply type:
+
 ```
 $ curl localhost:4000/users
```
Comment (craicoverflow) [inline readme.md:199]: You should also ad the output in CURL commands. So for example:

```sh
$ curl localhost:4000/ping
{
   "pong": true
}
```

this should help the reader visualise what happens and they will know if their command is actually doing what is expected.

File: readme.md [hunk 2]
```diff
@@ -174,83 +192,101 @@ app.get('/users', (req, res) => {
...
+
 ```
 $ curl localhost:4000/users/John
 ```
+
 See how cool REST can be?! Now, let's go through another HTTP method - DELETE! To do that add another method to our server:
+
 ```js
-app.delete('/users/delete/:userName', (req, res) => {
+app.delete('/users/:userName', (req, res) => {
```
Comment (craicoverflow) [inline readme.md:229]: With REST, you can have the exact same URL for GET, POST, DELETE, PUT etc. It is the HTTP protocol that dictates which endpoint your request hits.

File: readme.md [hunk 3]
```diff
@@ -174,83 +192,101 @@ app.get('/users', (req, res) => {
...
 ```
-- It returns all registered users already but to double check, you can run our curl command for getting all users also to see our database updated with an extra user!
 
-- Lastly, our server accepts only requests from localhost, which even although it is a very simple server, makes it unusable to anyone except ourselves. To change it and allow requests from other sources we need to enable CORS support. I highly recommend reading more about CORS [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS). To do that, first, add dependencies to our project:
-```
-$ npm install cors
+This returns all registered users already but to double check, you can run our curl command for getting all users also to see our mock database updated with an extra user!
+
+Finally, our server only allows requests from localhost, which even although it is a very simple server, makes it unusable to anyone except ourselves. To change it and allow requests from other sources we need to enable CORS support. I highly recommend reading more about CORS [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS). To do that, first, add dependencies to our project:
```
Comment (craicoverflow) [inline readme.md:265]: It might be nice to also explain what CORS is, briefly.

File: readme.md [hunk 4]
```diff
@@ -268,31 +304,30 @@ app.get('/users/:userName', (req, res) => {
...
 
+## Summary
+
+There you go! Your very first Node.js server that uses Express with REST endpoints. It is very easy, there's a lot more you can do with REST, but I guess my target here was to give you a brief idea of what REST is. What we just did is similar to what I was exposed to on my adventure to become a software developer, which is still ongoing 😉
+
+It was also a huge eye-opener for me as for understanding client-server architecture! However,as I'm sure you can imagine now, you need to expose a different endpoint for each data you want to send, our project has only 2 entities, users and tasks, and it's very plain and simple - imagine how complex REST could be if you add another 10, 20 or more entities which can somehow interact with each other!
+
+Imagine that everything we did so far would need to be accessing the database through, for example, SQL statements, that would need to be placed in a different layer of our server to hide business logic! Imagine writing all those REST endpoints for your "dream" application! Not that easy anymore, is it?!
```
Comment (craicoverflow) [inline readme.md:332]: I'd recommend adding some links to useful resources that you would recommend to learn more. What did you use to learn REST?

**Review by MStokluska — APPROVED** [review]
great suggestions! Thanks @craicoverflow

**Comment by craicoverflow** [comment]
> great suggestions! Thanks @craicoverflow

be sure to update the actual code of your server in this project to match!

**Pull request merged**